tests/unit/test_salesforce.py

```diff
@@ -0,0 +1,101 @@
+import pandas as pd
+import pytest
+
+from viadot.sources import Salesforce
+
+TABLE_TO_DOWNLOAD = "Account"
+TABLE_TO_UPSERT = "Contact"
+TEST_LAST_NAME = "prefect-viadot-test"
+ID_TO_UPSERT = "0035E00001YGWK3QAP"
+
+
+@pytest.fixture(scope="session")
+def salesforce():
+    s = Salesforce(config_key="sales_force_dev")
+    yield s
+
+
+@pytest.fixture(scope="session")
+def test_df_data(salesforce):
+    data = {
+        "Id": [ID_TO_UPSERT],
+        "LastName": [TEST_LAST_NAME],
+    }
+    df = pd.DataFrame(data=data)
+
+    yield df
+
+    data_restored = {
+        "Id": [ID_TO_UPSERT],
+        "LastName": ["LastName"],
+    }
+    df_restored = pd.DataFrame(data=data_restored)
+    salesforce.upsert(df=df_restored, table=TABLE_TO_UPSERT)
+
+
+@pytest.fixture(scope="session")
+def test_df_external(salesforce):
+    data = {
+        "LastName": [TEST_LAST_NAME],
+        "SAPContactId__c": ["111"],
+    }
+    df = pd.DataFrame(data=data)
+    yield df
+
+    data_restored = {
+        "Id": [ID_TO_UPSERT],
+        "LastName": ["LastName"],
+    }
+    df_restored = pd.DataFrame(data=data_restored)
+    salesforce.upsert(df=df_restored, table=TABLE_TO_UPSERT)
+
+
+def test_upsert_empty(salesforce):
+    try:
```

Comment:
Can you add a docstring here explaining this tests that an exception is not raised?

Reply:
I will add, but I think I will remove this test because it doesn't make much sense. Because in other upserts there is a try exception block that will catch the error

Reply:
[✅ 🔥 Removed test test_upsert_empty](https://github.com/dyvenia/viadot/pull/629/commits/ab1d1af708b47241f6ebaefa05311edb616c1bcb)